api/client.go

```diff
@@ -168,41 +165,85 @@ type readsResponseHeader interface {
 	setValuesFromHeader(header http.Header) error
 }
 
-func get[Res any](client Client, path string, query Query) (*Res, error) {
-	return request[EmptyRequest, Res](client, http.MethodGet, path, query, nil)
+func get[Res any](ctx context.Context, client Client, path string, query Query) (*Res, error) {
+	req, err := client.buildRequest(ctx, http.MethodGet, path, query, nil)
+	if err != nil {
+		return nil, err
+	}
+	return request[Res](client, req)
 }
 
 func post[Req, Res any](client Client, path string, req *Req) (*Res, error) {
-	return request[Req, Res](client, http.MethodPost, path, Query{}, req)
+	body, err := encodeRequestBody(req)
+	if err != nil {
+		return nil, err
+	}
+	ctx := context.TODO()
+	httpReq, err := client.buildRequest(ctx, http.MethodPost, path, nil, body)
+	if err != nil {
+		return nil, err
+	}
+	return request[Res](client, httpReq)
+}
+
+func encodeRequestBody(req any) (io.Reader, error) {
+	b, err := json.Marshal(req)
```

Comment:
I think we should keep the nil check here, it's valid to make a post request with no body:
```
if req != nil {
		b, err := json.Marshal(req)
		if err != nil {
			return nil, fmt.Errorf("marshal request: %w", err)
		}

		return bytes.NewReader(b), nil
	}
return nil, nil // not sure if this would cause issues elsewhere
```

Reply:
I've added the nil check here